### ISSUE TITLE
long recording file with chemogenetic inhibition 

### ISSUE THREAD

**Issue by smartbenjamin** [MouseLand/Kilosort#479]
I have a long recording file (~5 hours, 32-channel) with chemogenetic inhibition (PSEM) of neuronal spikes in the final 3-hours. One of my goals is to check the efficacy of chemo-genetic inhibition (and it works very well). When I just sort the spike in the first hour, I could see many units appeared in Phy GUI. However, when I sort the entire file, many of these units just disappear in Phy GUI. The reason might be the units are not detected in more than half of the recording files. I am wondering, would it be possible that just let kilosort (v2.5) learn the templates in the first 2 hours, and apply them to the rest of the data? Or, any other strategies that might work. Thanks!

**Comment by agbondy** [comment]
In your Kilosort config file, you could try lowering the value of ops.minFR to near or equal to zero. The default value is 1/50 which means that if a template's spike rate drops below that value for some amount of time, the template is removed. This may be the behavior that is causing the issue you are having. Of course, setting ops.minFR=0 means you may get some junk clusters with only a tiny handful of spikes assigned to them, but you can always remove those post hoc.

**Comment by smartbenjamin** [comment]
Thank you very much, agbondy!! I did try to lower down the ops.minFR into 0.01 or 0 Hz. However, most of the units are basically noises, the good units are disappeared (probably due to the final merge and split procedures...). 

I also did some tests on the 'learnTemplates.m' under the mainLoop folder to restrain the learning algorithm to the first 2 hours (5 hours in total) of the recording data (by inserting "**niter = niter *2/5**" )in line 98 before ibatch loop. It works much better. I am not sure whether it is a good way to do it so far. By checking "runTemplate.m", it seems it's possible to sort spikes based on previously defined templates, are you familiar with it?

Thank you!
